url/tools/IdnaTestV2-parser.py

```diff
@@ -92,13 +96,11 @@ def parse(lines, exclude_ipv4_like, exclude_std3, exclude_bidi):
                 continue
 
         if exclude_std3:
-            if re.search(r"\u2260|\u226E|\u226F|\<|\>|\$|,", to_unicode):
+            if re.search(r"\<|\>|\$|\?|,", to_unicode):
```

Comment:
* I removed  `\u2260`, `\u226E` and `\u226F`. because they are labeled `valid` starting with UTS46 15.1.0.
* I added `?` because there are tests with it and it is the [forbidden domain code point](https://url.spec.whatwg.org/#forbidden-domain-code-point).
* I am not sure if `$` and `,` are still needed here?

Reply:
Filtering `?` is the cause for many removed tests. I think I'd rather keep them. Though if they now have a non-null expected output there's a problem.

Reply:
`IdnaTestV2.txt` contains many tests labeled only with `[U1]` and containing `?`, e.g.
```
xn--88e732c.xn--1-?-pzc; ≮ᡬ.σ1-?; [U1]; xn--88e732c.xn--1-?-pzc; ; ;  # ≮ᡬ.σ1-?
```
So they will have non-null output because we are ignoring `U1`,

Reply:
My tests show that we can safely remove `$` and `,` from here. Tests that contain them now have correct labels. This adds 47 new test cases.

Reply:
I should have documented rationale here. I suspect `,` is disallowed because https://github.com/whatwg/url/issues/390 is unresolved. I can't find anything for `$` though. (https://github.com/whatwg/url/issues/159 seems too old.)

Anyway, I would be okay with allowing them. We can always update the tests later.

For `?` I guess we have to disallow it then, but that still leaves a number of removed tests. I guess we can just add a parallel file for that and we just run the compare script each time we want to update this file to see if we need to add more "discarded" tests.

Reply:
I found a mention of the `$` in https://github.com/whatwg/url/issues/744
> The test for ToASCII("$") is marked P1 and V6, not U1. ...

But now in UTS46 16.0.0 it is correctly marked as `U1`.

Reply:
Nice!